packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -0,0 +1,107 @@
+import { RegionSubmenu, RegionSubmenuResponse } from '../../../shared/ui/common/regionSubmenu'
+import { CancellationError } from '../../../shared/utilities/timeoutUtils'
+import { LogStreamFilterResponse, LogStreamFilterSubmenu } from '../liveTailLogStreamSubmenu'
+import { getLogger } from '../../../shared'
+
+const localize = nls.loadMessageBundle()
+
+export interface TailLogGroupWizardResponse {
+    regionLogGroupSubmenuResponse: RegionSubmenuResponse<string>
+    logStreamFilter: LogStreamFilterResponse
+    filterPattern: string
+}
+
+export async function tailLogGroup(logData?: { regionName: string; groupName: string }): Promise<void> {
+    const wizard = new TailLogGroupWizard(logData)
+    const wizardResponse = await wizard.run()
+    if (!wizardResponse) {
+        throw new CancellationError('user')
+    }
+
+    //TODO: Remove Log. For testing while we aren't yet consuming the wizardResponse.
+    getLogger().info(JSON.stringify(wizardResponse))
+}
+
+export class TailLogGroupWizard extends Wizard<TailLogGroupWizardResponse> {
+    public constructor(logGroupInfo?: CloudWatchLogsGroupInfo) {
+        super({
+            initState: {
+                regionLogGroupSubmenuResponse: logGroupInfo
+                    ? {
+                          data: logGroupInfo.groupName,
+                          region: logGroupInfo.regionName,
+                      }
+                    : undefined,
+            },
+        })
+        this.form.regionLogGroupSubmenuResponse.bindPrompter(createRegionLogGroupSubmenu)
+        this.form.logStreamFilter.bindPrompter((state) => {
+            if (!state.regionLogGroupSubmenuResponse?.data) {
+                throw Error('LogGroupName is null')
+            }
+            return new LogStreamFilterSubmenu(
+                state.regionLogGroupSubmenuResponse.data,
+                state.regionLogGroupSubmenuResponse.region
+            )
+        })
+        this.form.filterPattern.bindPrompter((state) => createFilterPatternPrompter())
+    }
+}
+
+export function createRegionLogGroupSubmenu(): RegionSubmenu<string> {
+    return new RegionSubmenu(
+        getLogGroupQuickPickOptions,
+        {
+            title: localize('AWS.cwl.tailLogGroup.logGroupPromptTitle', 'Select Log Group to tail'),
+            buttons: [createExitButton()],
+        },
+        { title: localize('AWS.cwl.tailLogGroup.regionPromptTitle', 'Select Region for Log Group') },
+        'LogGroups'
+    )
+}
+
+async function getLogGroupQuickPickOptions(regionCode: string): Promise<DataQuickPickItem<string>[]> {
+    const client = new DefaultCloudWatchLogsClient(regionCode)
+    const logGroups = client.describeLogGroups()
+
+    const logGroupsOptions: DataQuickPickItem<string>[] = []
+
+    for await (const logGroupObject of logGroups) {
+        if (!logGroupObject.arn || !logGroupObject.logGroupName) {
+            throw Error('LogGroupObject name or arn undefined')
```

Comment:
I think we typically throw ToolkitErrors instead of regular errors: https://github.com/aws/aws-toolkit-vscode/blob/master/docs/arch_develop.md#exceptions. Theres a couple places here where it might make sense